library/CMakeLists.txt

```diff
@@ -129,11 +129,11 @@ if(WIN32)
     set(libs ${libs} ws2_32)
 endif(WIN32)
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    SET(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
-    SET(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
-    SET(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
-    SET(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
```

Comment:
Unfortunately because we are now matching by compiler, it does not make sense to set both C and C++ flags based on the C compiler - https://github.com/Mbed-TLS/mbedtls/pull/8757/ is more like what this should look like (but unfortuntely got closed)

Reply:
Ah hadn't seen that MR. Ideally we'd even check the `CMAKE_<LANG>_COMPILER_RANLIB` vendor, but CMake does not identify that. I'll update the MRs.